src/clr-angular/data/datagrid/providers/table-size.service.ts

```diff
@@ -33,4 +45,20 @@ export class TableSizeService {
     getColumnDragHeight(): string {
         return `${this.tableRef.clientHeight}px`;
     }
+
+    private subscriptions: Subscription[] = [];
+
+    ngOnDestroy() {
+        this.subscriptions.forEach(sub => sub.unsubscribe());
+    }
+
+    updateRowWidth(): void {
+        let newWidth: number = 0;
+        this.renderer.removeStyle(this.tableRef, "width");
+        this.columns = Array.from(this.tableRef.querySelectorAll(".datagrid-column"));
+        this.columns.forEach(item => {
+            newWidth += item.clientWidth;
+        });
+        this.renderer.setStyle(this.tableRef, "width", newWidth + "px");
```

Comment:
I don't understand where this is needed at the moment. Is this another inert artifact from pinned columns? If so, that one actually sets a hard width on the datagrid so it's not as inert as we would like, we probably want it out of this PR.

Or I forgot about something you explained to me some time ago and it is needed. 🤔

Reply:
This is in there because of how hovering and the expandable row columns is behaving. Without an explicit width, the content / hover area is limited to whats visible even when the columns are wide enough for overflow on the x-axis.